internal/graph/rate_limited_check_resolver.go

```diff
@@ -0,0 +1,166 @@
+package graph
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/openfga/openfga/internal/build"
+	"github.com/openfga/openfga/pkg/telemetry"
+)
+
+// RateLimitedCheckResolverConfig encapsulates configuration for rate limited check resolver
+type RateLimitedCheckResolverConfig struct {
+	TimerTickerFrequency time.Duration
+	MediumPriorityLevel  uint32
+	MediumPriorityShaper uint32
+	LowPriorityLevel     uint32
+	LowPriorityShaper    uint32
+}
+
+// RateLimitedCheckResolver will prioritize high priority queue (requests with smaller number of dispatches) over
+// medium and low priority queue (requests with larger number of dispatches)
```

Comment:
This is the doc I was looking for :) 

why do we need two queues to represent `(requests with larger number of dispatches)`?